### ISSUE TITLE
[Autocomplete] Create a base EntityAutocompleteType

### ISSUE THREAD

**Issue by bastien70** [symfony/ux#774]
Hey!

I need for my `EntityType` to have the **autocomplete** but with **Ajax**.

But I don't want to have to create a type every time to use it.

My goal would therefore be to be able to keep the same configuration that we use for an `EntityType` (with the 'class', '`query_builder`' options) but with **Ajax**.

I saw this ticket that floated the idea of ​​an '`ajaxify`' option: https://github.com/symfony/ux/issues/433

But I'm rather thinking of creating a new type which would simply be the "`EntityAutocompleteField`", which would take, as said, exactly the same configuration as the `EntityType`, and which, thanks to the **autocomplete**, would add **Ajax**.

One of the reasons why I also want this type of file is that when we use **LiveComponent** and our form has events with show/hide fields, with the `EntityType` we unfortunately lose the autocomplete of base. While with the **Symfony UX Autocomplete**, the field is recreated each time and initialized with **TomSelect**.

The problem is that, if I create a `EntityAutocompleteField` like this : 

```php
#[AsEntityAutocompleteField]
class EntityAutocompleteField extends AbstractType
{
    public function configureOptions(OptionsResolver $resolver)
    {
        $resolver->setDefaults([
            'class' => null,
        ]);
    }

    public function getParent(): string
    {
        return ParentEntityAutocompleteType::class;
    }
}
```

I can't set '`class`' option to null here.

My goal will be to do this from my Form : 

```php
            $builder
                ->add('lastDiplomaAttended', EntityAutocompleteField::class, [
                    'class' => DiplomaType::class,
                    'label' => 'Dernier diplôme préparé',
                    'placeholder' => 'Veuillez choisir un diplôme',
                    'query_builder' => fn (DiplomaTypeRepository $r) => $r
                        ->createQueryBuilder('d')
                        ->innerJoin('d.enrollmentPlan', 'e')
                        ->andWhere('e.code = :enrollmentPlan')
                        ->setParameter('enrollmentPlan', $contract->getCurrentStudentRegistration()->getEnrollmentPlan()->getCode())
                        ->orderBy('d.code', 'ASC'),
                ])
```

In this case, the '`class`' option doesn't replace the `EntityAutocomplete` class option. So, there will be no results. And so, the '`query_builder`' option is not really available here, we have to use a '`filter_query`'

**Comment by WebMamba** [comment]
Maybe try to ask on the support channel of the symfony slack. You gonna get more help for sure
😁

**Comment by weaverryan** [comment]
Hi!

> One of the reasons why I also want this type of file is that when we use LiveComponent and our form has events with show/hide fields, with the EntityType we unfortunately lose the autocomplete of base. While with the Symfony UX Autocomplete, the field is recreated each time and initialized with TomSelect.

That will be fixed in Live Components 2.8 🔥 

> The problem is that, if I create a EntityAutocompleteField like this :

```
            $builder
                ->add('lastDiplomaAttended', EntityAutocompleteField::class, [
                    'class' => DiplomaType::class,
                    'label' => 'Dernier diplôme préparé',
                    'placeholder' => 'Veuillez choisir un diplôme',
                    'query_builder' => fn (DiplomaTypeRepository $r) => $r
                        ->createQueryBuilder('d')
                        ->innerJoin('d.enrollmentPlan', 'e')
                        ->andWhere('e.code = :enrollmentPlan')
                        ->setParameter('enrollmentPlan', $contract->getCurrentStudentRegistration()->getEnrollmentPlan()->getCode())
                        ->orderBy('d.code', 'ASC'),
                ])
````


Yea, I understand the situation. The reason that the custom class is required is that, when the Ajax request is made for the "options", we need to be able to reload the original configuration for that field (e.g. which class, which query builder, etc). When you can pass custom options to your autocomplete field in the middle of a form class, when the Ajax call is made, we have no way to know how/where to load those same options. The custom class solves that. The AJAX request is made to a URL like `/autocomplete/diploma_attended` where `diploma_attended` is a string that is mapped to your custom class (e.g. `DiplomaAttendedAutocompleteField`, which we can then load and ask for all of your options.

But this DOES introduce some limitations. Those are discussed over on: https://github.com/symfony/ux/issues/391

I'm going to close this issue in favor of that one - but if I've misunderstood anything, let me know. And hopefully Live Components 2.8 (should be ready soon) will make your life easier!

Cheers!
